src/sdk/model/user/user.ts

```diff
@@ -199,6 +201,31 @@ export class User {
     });
   }
 
+  /**
+   * Update the user.
+   * @param {UserUpdateRequest} request
+   * @returns {Promise<User>}
+   */
+  async update(request: UserUpdateRequest): Promise<User> {
+    return Iland.getHttp().put(`/users/${this.username}`, request.json).then((response) => {
+      const json = response.data as UserJson;
+      return new User(json);
+    });
+  }
+
+  /**
+   * Delete the current user.
+   * @returns {Promise<void>}
+   * @throws Error
+   */
+  async delete(): Promise<void> {
+    return Iland.getHttp().delete(`/users/${this.username}`).then(() => {
+      return;
+    }).catch(() => {
+      throw new Error(`Something went wrong with user deletion.`);
```

Comment:
what is this?

Reply:
Handling the case something went wrong with user delete API side. If it return a 4xx or 5xx request. The error message can be changed if you want.

Reply:
theres no reason to catch the error, we should let the api bubble up like we do everywhere else. It could give the user some interesting details about the reason. No reason for the success promise handler that just returns either.